### DISCUSSION TITLE
issues using npm

### DISCUSSION THREAD

**Discussion by dderjoel** [swimauger/tsc-hooks#17] [Q&A]
Hi,
you mentioned [here](https://github.com/microsoft/TypeScript/issues/30835#issuecomment-850098727), that you have had issues with npm. Which issues did you have using npm?

I installed tsc-hooks with npm, did `tsc -p .` and it copied as expected.
I updated my project (among others typescript from 4.3.5 to 4.4.2) , cleaned `dist` and now its not working anymore.
I undid the updates from earlier, but it is still not working...

Was that similar to your issues?

I  run into other issues (unrelated to the main projects and tsc-hook) when using yarn, so I could not test it the issue from above would be solved using yarn...

**Comment by swimauger** [comment]
Hey,
Yes, so that's actually a different issue. The problem you brought up happens because tsc-hooks does an injection into the typescript modules' `tsc` cli on post installation. When you installed typescript again it overwrote that injection. If you were to install tsc-hooks again, it would most likely correct the problem... Unfortunately the cleanest solution I see so far for this is adding another injection, but into `npm` or `yarn`, which I'm not particularly sold on.

Now, the issue, I brought up about installing from `npm` globally. I believe my issue was I use `nvm`, but installed NodeJS directly from the web before that. Creating two `tsc`'s on Mac, /usr/local/bin/tsc and .nvm/versions/node/v15.14.0/lib/node_modules/typescript/bin/tsc

**Reply by dderjoel** [reply]
I see. Okay, then we should add those two things to the main readme page.
Caution@mac-users, ...
Caution@ everybody: "when updating typescript, reinstall `tsc-hooks`.

Or, for the latter, now that I am a bit more familiar with the code, `node run postinstall` could be enough.
